doc/source/whatsnew/v0.23.0.rst

```diff
@@ -105,6 +105,16 @@ Please note that the string ``index`` is not supported with the round trip forma
 .. ipython:: python
    :okwarning:
 
+   df = pd.DataFrame(
+       {
+           'foo': [1, 2, 3, 4],
+           'bar': ['a', 'b', 'c', 'd'],
+           'baz': pd.date_range('2018-01-01', freq='d', periods=4),
+           'qux': pd.Categorical(['a', 'b', 'c', 'c'])
+       },
+       index=pd.Index(range(4), name='idx')
+   )
+
    df.index.name = 'index'
```

Comment:
I don't understand how this worked on main, but it seems to fix the docs build issue here.